hamocc/mo_apply_oafx.F90

```diff
@@ -0,0 +1,102 @@
+! Copyright (C) 2021  J. Schwinger
+!
+! This file is part of BLOM/iHAMOCC.
+!
+! BLOM is free software: you can redistribute it and/or modify it under the
+! terms of the GNU Lesser General Public License as published by the Free 
+! Software Foundation, either version 3 of the License, or (at your option) 
+! any later version. 
+!
+! BLOM is distributed in the hope that it will be useful, but WITHOUT ANY 
+! WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS 
+! FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for
+! more details. 
+!
+! You should have received a copy of the GNU Lesser General Public License 
+! along with BLOM. If not, see https://www.gnu.org/licenses/.
+
+
+module mo_apply_oafx
+!******************************************************************************
+!
+!   J.Schwinger             *NORCE Climate, Bergen*             2021-11-15
+!
+! Modified
+! --------
+!
+! Purpose
+! -------
+!  -Routines for applying ocean alkalinization
+!
+!
+! Description:
+! ------------
+!
+!  -subroutine alkalinization
+!     Apply alkalinization to the top-most model layer.
+!
+!
+!******************************************************************************
+  implicit none
+
+  private
+  public :: apply_oafx
+
+!******************************************************************************
+contains
+
+
+
+subroutine apply_oafx(kpie,kpje,kpke,pddpo,omask,oafx)
+!******************************************************************************
+!
+!     J. Schwinger            *NORCE Climate, Bergen*     2021-11-15
+!
+! Purpose
+! -------
+!  -apply alkalinization to the top-most model layer.
+!
+! Changes: 
+! --------
+!
+!
+! Parameter list:
+! ---------------
+!  *INTEGER*   *kpie*    - 1st dimension of model grid.
+!  *INTEGER*   *kpje*    - 2nd dimension of model grid.
+!  *REAL*      *pddpo*   - size of grid cell (depth) [m].
+!  *REAL*      *omask*   - land/ocean mask (1=ocean)
+!  *REAL*      *oafx*    - alkalinization field to apply [kmol m-2 yr-1]
+!
+!******************************************************************************
+  use mo_control_bgc, only: dtb,do_oalk
+  use mo_carbch,      only: ocetra
+  use mo_param1_bgc,  only: ialkali
+
+  implicit none
+
+  integer, intent(in) :: kpie,kpje,kpke
+  real,    intent(in) :: pddpo(kpie,kpje,kpke)
+  real,    intent(in) :: omask(kpie,kpje)
+  real,    intent(in) :: oafx(kpie,kpje)
+
+  ! local variables 
+  integer :: i,j
+
+  if (.not. do_oalk) return 
+
+  ! alkalinization in topmost layer 
+  do j=1,kpje
+  do i=1,kpie
+    if (omask(i,j).gt.0.5) then
+      ocetra(i,j,1,ialkali)=ocetra(i,j,1,ialkali)+oafx(i,j)*dtb/365./pddpo(i,j,1)
```

Comment:
No, we don't have that. We don't even have a module where such parameters would fit in naturally  (both mo_control and mo_param1 are not really the place for general purpose parameters). Should we make a new on (e.g. mo_param0)?

Reply:
I guess, I was also thinking about leap years (which might have been defined in BLOM)  and/or other setups for more crazy applications (longer/shorter days, years, ...). Technically, it would be good to have such module, however, I see for time-related parameters the need to keep these manageable via the driving model (aka BLOM for the most time). Other than that hamocc_init is currently the place, where I would store/define it. But nevermind, it was a really general question.

Reply:
I agree, with respect to calendar related things we would need to pass this from BLOM to HAMOCC via hamocc_ini (although I'm not sure whether we would ever run NorESM or BLOM on a calendar other than 'noleap'). I'm closing and merging this now - thanks for the thorough reviews!